backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -399,20 +595,35 @@ public Response updateBehaviorGroupActions(@Context SecurityContext sec,
     @Consumes(APPLICATION_JSON)
     @Produces(TEXT_PLAIN)
     @Operation(summary = "Update the list of behavior groups for an event type", description = "Updates the list of behavior groups associated with an event type.")
-    @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
     @APIResponse(responseCode = "200", content = @Content(schema = @Schema(type = SchemaType.STRING)))
     @Transactional
     public Response updateEventTypeBehaviors(@Context SecurityContext sec,
                          @Parameter(description = "UUID of the eventType to associate with the behavior group(s)") @PathParam("eventTypeId") UUID eventTypeId,
                          @Parameter(description = "Set of behavior group ids to associate") Set<UUID> behaviorGroupIds) {
+        if (this.backendConfig.isKesselBackendEnabled()) {
+            this.kesselAuthorization.hasPermissionOnResource(sec, WorkspacePermission.NOTIFICATIONS_WRITE, ResourceType.WORKSPACE, WORKSPACE_ID_PLACEHOLDER);
+            behaviorGroupIds.forEach(id -> this.kesselAuthorization.hasPermissionOnResource(sec, ResourcePermission.VIEW, ResourceType.BEHAVIOR_GROUP, id.toString()));
```

Comment:
```suggestion
            behaviorGroupIds.forEach(id -> this.kesselAuthorization.hasPermissionOnResource(sec, ResourcePermission.WRITE, ResourceType.BEHAVIOR_GROUP, id.toString()));
```

Reply:
I doubted about this one. Since we are simply adding a behavior group to an event type, but not really editing the behavior group, should not we just make sure that the user can see that behavior group? :thinking: